### PR TITLE
Replace lib/utilities/DAG.js with dag-map package

### LINKED ISSUE #6881: Replace DAG utility with dag-map
The original source for [lib/utilities/DAG.js](https://github.com/ember-cli/ember-cli/blob/6ff2aa283c3961b6bf8f7981f3453d1db35afa9b/lib/utilities/DAG.js) was from Ember itself. Extracting it for use in ember-cli led to us extracting form Ember into [dag-map](https://github.com/krisselden/dag-map) package.

The supported API's are roughly the same, and hopefully removing the internal version in favor of dag-map package is straight forward.

### PR DESCRIPTION
**Pull request by rwwagner90**
Resolves #6881

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -129,6 +129,7 @@
     "chai-as-promised": "^6.0.0",
     "chai-files": "^1.2.0",
     "co": "^4.6.0",
+    "dag-map": "^2.0.2",
```
Comment (rwjblue) [inline package.json]: This should be a dependency, not a devDep (it is used for normal app runtime).

Comment (rwwagner90) [inline package.json]: Sorry about that, just moved to deps!

**Comment by Turbo87** [comment]
@homu r+

**Comment by Turbo87** [comment]
@homu retry

@rwjblue @stefanpenner we seem to have introduced quite a bit of flakiness to our tests again... 🤔

**Comment by Turbo87** [comment]
@homu retry 😠

**Pull request merged**